### PR TITLE
GEODE-5604 Update gradle syntax to comply with 5.0

### PR DESCRIPTION
**Pull request by rhoughton-pivot**
* Fix subproject names for gradle 5.0
** stop using '/' and use preferred ':' character for nested projects
* Update performance profiling plug-in to be Gradle 5.0 compliant
* Fix input/output warnings on geode-old-versions
* Don't use custom configuration in geode-modules-assembly
* All test configurations inherit annotationProcessor from mainSourceSet

Co-authored-by: Dick Cavender <dcavender@pivotal.io>
Co-authored-by: Jacob Barrett <jbarrett@pivotal.io>
Co-authored-by: Patrick Rhomberg <prhomberg@pivotal.io>
Co-authored-by: Robert Houghton <rhoughton@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by rhoughton-pivot** [comment]
Do not accept until closed-side changes are ready

**Comment by metatype** [comment]
Have you reviewed the effect this will have on the POM?  Can you post that as a comment?

Right now I'm -1 based on what appears to be broken pom dependencies.  For example, geode-cq *must* expose a dependency on geode-core.

**Comment by metatype** [comment]
What do you mean by "closed side changes"??

File: gradle/test.gradle [hunk 1]
```diff
@@ -73,9 +73,11 @@ subprojects {
     }
     // Because EnsureCorrectRunsWithProcessor needs access to
     // CategoryWithParameterizedRunnerFactory. The specific version of geode-junit is not important.
-    apt(group: 'org.apache.geode', name: 'geode-junit', version: '1.3.0') {
-      transitive = false
-    }
+    apt project(':geode-junit')
+//        {
```
Comment (pivotal-jbarrett) [inline gradle/test.gradle]: Delete!

File: gradle/test.gradle [hunk 2]
```diff
@@ -164,13 +176,13 @@ subprojects {
 
     outputs.upToDateWhen { false }
 
-    classpath += project.sourceSets.distributedTest.runtimeClasspath
+//    classpath += project.sourceSets.distributedTest.runtimeClasspath
```
Comment (pivotal-jbarrett) [inline gradle/test.gradle]: Delete

File: buildSrc/build.gradle [hunk 3]
```diff
@@ -21,7 +21,16 @@ repositories {
 }
 
 dependencies {
-  compile group: 'org.apache.geode', name: 'geode-junit', version: '1.3.0'
+  implementation(group: 'org.apache.geode', name: 'geode-junit', version: '1.3.0') {
```
[the remark this answers is not in the document]
Comment (rhoughton-pivot) [reply buildSrc/build.gradle]: If we don't exclude some of these, then we get warnings from buildSrc about the annotation processors again

**Comment by pivotal-jbarrett** [comment]
@rhoughton-pivot @PurelyApplied Rebase with the latest develop and run the `geode-assembly:integrationTest` target. It tests a few dependency checks in jars and assembly.

**Pull request merged**